commons/all/src/main/java/org/infinispan/commons/io/FileWatcher.java

```diff
@@ -0,0 +1,115 @@
+package org.infinispan.commons.io;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.commons.util.Util;
+
+/**
+ * @since 15.0
+ */
+public class FileWatcher implements Runnable {
+   private static final Log log = LogFactory.getLog(FileWatcher.class);
+
+   private static FileWatcher INSTANCE;
+
+   private final Thread thread;
+   private final WatchService watchService;
```

Comment:
This won't work as expected in K8s:

https://blog.arkey.fr/2019/09/13/watchservice-and-bind-mount/

Reply:
I've modified it to use plain last modified time